hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CustomizedScanInfoBuilder.java

```diff
@@ -80,4 +97,23 @@ public KeepDeletedCells getKeepDeletedCells() {
     return keepDeletedCells != null ? keepDeletedCells : scanInfo.getKeepDeletedCells();
   }
 
+  @Override
+  public int getMinVersions() {
+    return minVersions != null ? minVersions : scanInfo.getMinVersions();
+  }
+
+  @Override
+  public void setMinVersions(int minVersions) {
+    this.minVersions = minVersions;
+  }
+
+  @Override
+  public Scan getScan() {
+    try {
+      return new Scan(scan);
```

Comment:
Might be good at some point to have an unmodifiable subclass or wrapper of Scan, so that any modification would cause an error. (not in this PR, though)

Reply:
+1 to Immutable Scan wrapper with unmodifiable getters

Reply:
+1..  That was something came to my mind too while reading the Jira desc. :-)

Reply:
Already the scan object been cloned while setting in this class..  Every time with get also we need to clone?

Reply:
@anoopsjohn  In the original draft it was only cloned in the constructor. @apurtell pointed out that that meant that one coproc override could call getScan(), call a setter on the Scan, and that a subsequent override would see the modified Scan, so he asked me to fix that.

Reply:
Ok.  An ImmutableScan would have helped. Fine. As long as we dont have lets do this.